library/Translate/src/Juvix/Contextify/Environment.hs

```diff
@@ -0,0 +1,394 @@
+module Juvix.Contextify.Environment
+  ( ErrorS (..),
+    ErrS,
+    SexpContext,
+    HasClosure,
+    passContextSingle,
+    passContext,
+    Pass (..),
+    extractInformation,
+    lookupPrecedence,
+  )
+where
+
+import Control.Lens hiding ((|>))
+import qualified Juvix.Core.Common.Closure as Closure
+import qualified Juvix.Core.Common.Context as Context
+import qualified Juvix.Core.Common.NameSpace as NameSpace
+import qualified Juvix.FrontendContextualise.InfixPrecedence.ShuntYard as Shunt
+import Juvix.Library
+import qualified Juvix.Library.NameSymbol as NameSymbol
+import qualified Juvix.Library.Sexp as Sexp
+import Prelude (error)
+
+-- Will make things more export friendly when the above code is gone
+
+-- TODO ∷ make this a standard data structure
+data ErrorS
+  = CantResolve [Sexp.T]
+  | UnknownSymbol NameSymbol.T
+  | ImpossibleMoreEles
+  | Clash
+      (Shunt.Precedence Sexp.T)
+      (Shunt.Precedence Sexp.T)
+  deriving (Show, Eq)
```

Comment:
New Error Types

Reply:
Is that a typo? Should it be `ImpossibleMoreElse`?

Reply:
Not a typo, it's impossible more elements